src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/FindArrayStatistics.cpp

```diff
@@ -17,121 +17,192 @@ template <typename T>
+          {
+            stdDevVal = meanValue;
+          }
+          else
+          {
+            std::vector<float64> difference;
+            difference.reserve(numTuples);
+            for(usize i = 0; i < numTuples; ++i)
+            {
+              if(m_Mask != nullptr && !m_Mask->isTrue(i))
+              {
+                continue;
+              }
+              if((*m_FeatureIds)[i] == static_cast<int32>(j))
+              {
+                difference.push_back(static_cast<float64>(m_Source[i] - meanValue));
+              }
+            }
+            const float64 squaredSum = std::inner_product(std::cbegin(difference), std::cend(difference), std::cbegin(difference), 0.0);
+            stdDevVal = static_cast<float32>(std::sqrt(squaredSum / static_cast<float64>(difference.size())));
+          }
+        }
+        m_StdDevArray->initializeTuple(j, stdDevVal);
       }
-      if(m_Histogram)
+      if(m_Histogram && histDataStore != nullptr)
       {
-        auto* arr7DataStore = array7->getDataStore();
-        if(arr7DataStore != nullptr)
+        std::vector<float32> histogram(m_NumBins, 0);
+        if(length[localFeatureIndex] > 0)
         {
-          std::vector<float32> vals = StaticicsCalculations::findHistogram(m_FeatureDataMap[i], m_HistMin, m_HistMax, m_HistFullRange, m_NumBins);
-          arr7DataStore->setTuple(i, vals);
+          float32 histMin = 0.0f;
+          float32 histMax = 0.0f;
+          if(m_HistFullRange)
+          {
+            histMin = static_cast<float32>(min[localFeatureIndex]);
+            histMax = static_cast<float32>(max[localFeatureIndex]);
+          }
+          else
+          {
+            histMin = m_HistMin;
+            histMax = m_HistMax;
+          }
+          const float32 increment = (histMax - histMin) / (m_NumBins);
+          if(std::fabs(increment) < 1E-10)
+          {
+            histogram = {static_cast<float32>(length[localFeatureIndex])};
+          }
+          else
+          {
+            for(usize i = 0; i < numTuples; ++i)
+            {
+              if(m_Mask != nullptr && !m_Mask->isTrue(i))
+              {
+                continue;
+              }
+              if((*m_FeatureIds)[i] == static_cast<int32>(j))
+              {
+                const auto value = static_cast<float32>(m_Source[i]);
+                const auto bin = static_cast<int32>((value - histMin) / increment); // find bin for this input array value
+                if((bin >= 0) && (bin < m_NumBins))                                 // make certain bin is in range
+                {
+                  ++histogram[bin]; // increment histogram element corresponding to this input array value
+                }
+                else if(value == histMax)
+                {
+                  histogram[m_NumBins - 1]++;
+                }
+              }
+            }
+          }
         }
+        histDataStore->setTuple(j, histogram);
       }
     }
   }
```

Comment:
```suggestion
            } // end of numTuples loop
          } // end of increment else
        } // end of length if
        histDataStore->setTuple(j, histogram);
      } // end of m_Histogram if
    } // end of outermost for loop
  } // end of compute
```
This is an optional change, but I think it helps readability with all the nested code blocks